src/main/java/client/util/CommandFactoryClient.java

```diff
@@ -0,0 +1,168 @@
+package client.util;
+
+// import client.ClientState; // NO LONGER NEEDED if parameter is removed
+import common.commands.*;
+import common.dto.HostGameRequestDTO;
+import common.dto.JoinPrivateGameRequestDTO;
+import common.dto.JoinPublicGameRequestDTO;
+import common.dto.UpdateDisplayNameRequestDTO;
+
+public class CommandFactoryClient {
+
+  private CommandFactoryClient() {}
+
+  /**
+   * Creates a Command DTO based on parsed user input.
+   *
+   * @param parsedData Parsed command name and arguments.
+   * @return A Command object or null if input is invalid/unknown.
+   */
+  public static Command createCommand(CommandParserClient.ParsedCommandData parsedData) {
+    if (parsedData == null || parsedData.commandName == null || parsedData.commandName.isEmpty()) {
+      return null;
+    }
+
+    String commandName = parsedData.commandName;
+    String arg = parsedData.getFirstArgument();
+
+    // --- Lobby & Session Setup Commands ---
+    switch (commandName) {
+      case "host game", "host case" -> {
```

Comment:
So you parse the same set of commands in game client, command factory client and command parser client

I bet you can simplify this and make this actually maintainable, because now if you want to add a command you have to make changes in THREE places, really hard to scale